ts-sdk/client/src/gpa/tokenBadge.ts

```diff
@@ -0,0 +1,41 @@
+import { GetProgramAccountsMemcmpFilter, getBase58Decoder, getAddressEncoder, Address, Account, GetProgramAccountsApi, Rpc } from "@solana/web3.js";
+import { TokenBadge, TOKEN_BADGE_DISCRIMINATOR, getTokenBadgeDecoder } from "../generated/accounts/tokenBadge";
+import { fetchDecodedProgramAccount } from "./utils";
+import { WHIRLPOOL_PROGRAM_ADDRESS } from "../generated/programs/whirlpool";
+
+export type TokenBadgeFilter = GetProgramAccountsMemcmpFilter & {
+  readonly __kind: unique symbol;
+}
+
+export function tokenBadgeWhirlpoolsConfigFilter(address: Address): TokenBadgeFilter {
+  return {
+    memcmp: {
+      offset: 8n, bytes: getBase58Decoder().decode(getAddressEncoder().encode(address)), encoding: "base58"
+    }
+  } as TokenBadgeFilter;
+}
+
+export function tokenBadgeTokenMintFilter(address: Address): TokenBadgeFilter {
+  return {
+    memcmp: {
+      offset: 40n, bytes: getBase58Decoder().decode(getAddressEncoder().encode(address)), encoding: "base58"
+    }
+  } as TokenBadgeFilter;
+}
+
+export async function fetchAllTokenBadgeWithFilter(rpc: Rpc<GetProgramAccountsApi>, ...filters: TokenBadgeFilter[]): Promise<Account<TokenBadge>[]> {
+  const discriminator = getBase58Decoder().decode(TOKEN_BADGE_DISCRIMINATOR);
+  const discriminatorFilter: GetProgramAccountsMemcmpFilter = {
+    memcmp: {
+      offset: 62n,
```

Comment:
hmm, 0n ?

Reply:
Good catch! Let me see if I can adjust the tests to that it also catches this filter